src/NServiceBus.Core.Tests/Msmq/MsmqQueueCreatorTests.cs

```diff
@@ -247,12 +229,16 @@ void DeleteQueueIfPresent(string queueName)
         {
             var path = MsmqAddress.Parse(queueName).PathWithoutPrefix;
 
-            if (!MessageQueue.Exists(path))
+            if (MessageQueue.Exists(path))
             {
-                return;
+                MessageQueue.Delete(path);
             }
+        }
 
-            MessageQueue.Delete(path);
+        string GetReallyLongQueueName()
+        {
+            var maxQueueNameForSetPermissionToWork = 102 - Environment.MachineName.Length;
```

Comment:
@andreasohlund - I get it now. After running more tests, it appears that 124 which is documented value is apparently a bit too much for the SetQueuePermissions and the call can blow. In my test, anything more than 114 character queue name will throw an exception on the SetQueuePermission. With this I will verify the implementation. I wish this was added as a code comment in the unit test to avoid this confusion.